### PR TITLE
Update javascript-quiz.md

### PR DESCRIPTION
**Pull request by TaylorOD**
- Add - [ ] for each question to make them easier to answer. JavaScript only has one answer it in.
- fix small formatting errors

### CODE REVIEW

File: javascript/javascript-quiz.md [hunk 1]
```diff
@@ -2,39 +2,39 @@ JavaScript Assessment
...
 
 #### Q1. Which operator returns true if the two compared values are not equal?
-- <>
-- ~
-- ==! 
-- !== <<<---Correct
+- [ ] <>
+- [ ] ~
+- [ ] ==! 
+- [ ] !== <<<---Correct
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:8]:
```suggestion
- [x] !==
```

File: javascript/javascript-quiz.md [hunk 2]
```diff
@@ -2,39 +2,39 @@ JavaScript Assessment
...
 
 #### Q2. How is a forEach statement different from a for statement?
-- Only a for statement uses a callback function
-- A for statement is generic, but a forEach statement can be used only with an array <<<--Correct
-- Only a forEach statement lets you specify your own iterator.
-- A forEach statement is generic, but a for statement ca be used only with an array
+- [ ] Only a for statement uses a callback function
+- [ ] A for statement is generic, but a forEach statement can be used only with an array <<<--Correct
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:12]:
```suggestion
- [x] A for statement is generic, but a forEach statement can be used only with an array
```

File: javascript/javascript-quiz.md [hunk 3]
```diff
@@ -526,24 +520,22 @@ class X {
 }
 var x = new X();
 ```
-- x.Y <<<<----Correct
-- x.Y()
-- x.get('Y')
-- x.get().Y
+- [ ] x.Y <<<<----Correct
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:523]:
```suggestion
- [x] x.Y
```

File: javascript/javascript-quiz.md [hunk 4]
```diff
@@ -514,10 +508,10 @@ var answer = 1;
 printA();
 printA();
 ```
-* 1 then 1
-* 1 then undefined
-* undefined the undefined <<<---CORRECT
-* undefined the 1
+- [ ] 1 then 1
+- [ ] 1 then undefined
+- [ ] undefined the undefined <<<---CORRECT
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:513]:
```suggestion
- [x] undefined the undefined
```

File: javascript/javascript-quiz.md [hunk 5]
```diff
@@ -489,21 +483,21 @@ console.log("hello");
...
 And instead of returning `undefined` the result => Object {}. 
 You can try it out open the console and check.
 
 ### Q55: Which collection object allows unique value to be inserted only once? 
-* Object
-* Set <<<---CORRECT
-* Array
-* Map
+- [ ] Object
+- [ ] Set <<<---CORRECT
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:498]:
```suggestion
- [x] Set
```

File: javascript/javascript-quiz.md [hunk 6]
```diff
@@ -489,21 +483,21 @@ console.log("hello");
 
 console.log(sayHello.prototype); 
 ```
-* undefined 
-* "hello"
-* an object with a constructor property <<<--- CORRECT
-* an error message
+- [ ] undefined 
+- [ ] "hello"
+- [x] an object with a constructor property <<<--- CORRECT
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:488]:
```suggestion
- [x] an object with a constructor property
```

File: javascript/javascript-quiz.md [hunk 7]
```diff
@@ -476,10 +470,10 @@ function setEnd() {
 setEnd();
 console.log(end); 
 ```
-* 10
-* 0
-* ReferenceError <<<---CORRECT 
-* undefined
+- [ ] 10
+- [ ] 0
+- [ ] ReferenceError <<<---CORRECT
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:475]:
```suggestion
- [x] ReferenceError 
```

File: javascript/javascript-quiz.md [hunk 8]
```diff
@@ -2,39 +2,39 @@ JavaScript Assessment
...
     return total * 1.05;
 }
 ```
-- addTax = 50;
-- return addTax 50;
-- addTax(50); <<<---Correct
-- addTax 50;
+- [ ] addTax = 50;
+- [ ] return addTax 50;
+- [ ] addTax(50); <<<---Correct
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:24]:
```suggestion
- [x] addTax(50);
```

File: javascript/javascript-quiz.md [hunk 9]
```diff
@@ -2,39 +2,39 @@ JavaScript Assessment
...
+- [ ] addTax 50;
 
 #### Q4. Which statement is the correct way to create a variable called rate and assign it the value 100?
-- let rate = 100; <<<---Correct
-- let 100 = rate;
-- 100 = let rate;
-- rate = 100;
+- [ ] let rate = 100; <<<---Correct
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:28]:
```suggestion
- [x] let rate = 100;
```

File: javascript/javascript-quiz.md [hunk 10]
```diff
@@ -2,39 +2,39 @@ JavaScript Assessment
...
+- [ ] let 100 = rate;
+- [ ] 100 = let rate;
+- [ ] rate = 100;
 
 #### Q5. Which statement creates a new object using the Person constructor?
-- var student = new Person(); <<<---Correct
-- var student = construct Person;
-- var student = Person();
-- var student = construct Person();
+- [ ] var student = new Person(); <<<---Correct
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:34]:
```suggestion
- [x] var student = new Person();
```

File: javascript/javascript-quiz.md [hunk 11]
```diff
@@ -57,54 +57,54 @@ console.log('Results shown');
...
    
-- `while (var i=1; i<=4; i++) {
+- [ ] `while (var i=1; i<=4; i++) {
     setTimeout(function() {
         console.log(i);
     }, i*1000);
    }`
    
-- `for (var i=1; i<=4; i++) {
+- [ ] `for (var i=1; i<=4; i++) {
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:74]:
```suggestion
- [x] `for (var i=1; i<=4; i++) {
```

File: javascript/javascript-quiz.md [hunk 12]
```diff
@@ -57,54 +57,54 @@ console.log('Results shown');
...
       {function(j) {
          setTimeout(function(){
            console.log(j);
          }, j*1000);
       })(i)
     }` <<<<----Correct
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:80]:
```suggestion
    }`
```

File: javascript/javascript-quiz.md [hunk 13]
```diff
@@ -57,54 +57,54 @@ console.log('Results shown');
...
      }, j*1000);
    }` 
    
 #### Q8. How does a function create a closure?
-- It reloads the document whenever the value changes.
-- It returns a reference to a variable in its parent scope <<<<---Correct
-- It completes execution without returning
-- It copies a local variable to the global scope
+- [ ] It reloads the document whenever the value changes.
+- [ ] It returns a reference to a variable in its parent scope <<<<---Correct
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:90]:
```suggestion
- [x] It returns a reference to a variable in its parent scope
```

File: javascript/javascript-quiz.md [hunk 14]
```diff
@@ -57,54 +57,54 @@ console.log('Results shown');
...
+- [ ] It completes execution without returning
+- [ ] It copies a local variable to the global scope
 
 #### Q9. Which statement creates a new function called discountPrice?
-- `let discountPrice = function(price) {
+- [ ] `let discountPrice = function(price) {
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:95]:
```suggestion
- [x] `let discountPrice = function(price) {
```

File: javascript/javascript-quiz.md [hunk 15]
```diff
@@ -57,54 +57,54 @@ console.log('Results shown');
...
      return price * 0.85;
     };` <<<<----Correct
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:97]:
```suggestion
    };`
```

File: javascript/javascript-quiz.md [hunk 16]
```diff
@@ -118,16 +118,16 @@ WinterStorm.prototype.precip = 'snow';
...
 console.log(bob.precip);
 ```
-- Storm()
-- undefined
-- 'rain'
-- 'snow' <<<---Correct
+- [ ] Storm()
+- [ ] undefined
+- [ ] 'rain'
+- [ ] 'snow' <<<---Correct
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:124]:
```suggestion
- [x] 'snow'
```

File: javascript/javascript-quiz.md [hunk 17]
```diff
@@ -118,16 +118,16 @@ WinterStorm.prototype.precip = 'snow';
...
 
 #### Q11. You need to match a time value such as 12:00:32. Which of the following regular expressions would work for your code?
-- /[0-9]{2,}:[0-9]{2,}:[0-9]{2,}/
-- /\d\d:\d\d:\d\d/ <<<---Correct, https://regex101.com/r/6e81up/2
-- /[0-9]+:[0-9]+:[0-9]+/ 
-- /  : : /
+- [ ] /[0-9]{2,}:[0-9]{2,}:[0-9]{2,}/
+- [ ] /\d\d:\d\d:\d\d/ <<<---Correct, https://regex101.com/r/6e81up/2
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:128]:
```suggestion
- [x] /\d\d:\d\d:\d\d/ https://regex101.com/r/6e81up/2
```

File: javascript/javascript-quiz.md [hunk 18]
```diff
@@ -138,32 +138,32 @@ function logThis() {
 }
 new logThis();
 ```
-- undefined
-- window
-- {desc: "logger"} <<<<---Correct
-- function
+- [ ] undefined
+- [ ] window
+- [ ] {desc: "logger"} <<<<---Correct
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:143]:
```suggestion
- [x] {desc: "logger"}
```

File: javascript/javascript-quiz.md [hunk 19]
```diff
@@ -138,32 +138,32 @@ function logThis() {
...
 let roadTypes = ['street', 'road', 'avenue', 'circle'];
 ```
-- roadTypes.2
-- roadTypes[3]
-- roadTypes.3
-- roadTypes[2] <<<---Correct
+- [ ] roadTypes.2
+- [ ] roadTypes[3]
+- [ ] roadTypes.3
+- [ ] roadTypes[2] <<<---Correct
```
Comment (Ebazhanov) [inline javascript/javascript-quiz.md:153]:
```suggestion
- [x] roadTypes[2]
```

**Review by Ebazhanov — CHANGES_REQUESTED** [review]
@TaylorOD thanks for the improvements I added some feedback for your please follow my suggestion and update the rest of the file, then we can merge it. Thank you in advance!

**Comment by TaylorOD** [comment]
This should be all set now! Thanks!

**Review by Ebazhanov — APPROVED** [review]
Yes, looks good now! Thank you @TaylorOD :punch:

**Pull request merged**